tests/gocase/unit/introspection/introspection_test.go

```diff
@@ -20,33 +20,230 @@
+		require.NoError(t, r.Err())
+		require.Equal(t, "OK", r.Val())
+		require.Regexp(t, ".*name=someothername.*", rdb.ClientList(ctx).Val())
+	})
+
+	t.Run("After CLIENT SETNAME, connection can still be closed", func(t *testing.T) {
+		func() {
+			c := srv.NewClient()
+			defer func() { require.NoError(t, c.Close()) }()
+			r := c.Do(ctx, "CLIENT", "SETNAME", "foobar")
+			require.NoError(t, r.Err())
+			require.Equal(t, "OK", r.Val())
+			require.Regexp(t, ".*name=foobar.*", rdb.ClientList(ctx).Val())
+		}()
+
+		// now the client should no longer be listed
+		require.Eventually(t, func() bool {
+			r := rdb.ClientList(ctx).Val()
+			return !strings.Contains(r, "foobar")
+		}, 5*time.Second, 100*time.Millisecond)
+	})
+
+	t.Run("Kill normal client", func(t *testing.T) {
+		defer func() { rdb = srv.NewClient() }()
+
+		c := srv.NewClient()
+		defer func() { require.NoError(t, c.Close()) }()
+		r := c.Do(ctx, "CLIENT", "SETNAME", "normal")
+		require.NoError(t, r.Err())
+		require.Equal(t, "OK", r.Val())
+		require.Regexp(t, ".*name=normal.*", rdb.ClientList(ctx).Val())
+
+		require.EqualValues(t, 1, rdb.ClientKillByFilter(ctx, "skipme", "yes", "type", "normal").Val())
+		require.EqualValues(t, 1, rdb.ClientKillByFilter(ctx, "skipme", "no", "type", "normal").Val())
+
+		// reconnect
+		rdb = srv.NewClient()
+		defer func() { require.NoError(t, rdb.Close()) }()
+
+		// now the client should no longer be listed
+		require.Eventually(t, func() bool {
+			r := rdb.ClientList(ctx).Val()
+			return !strings.Contains(r, "normal")
+		}, 5*time.Second, 100*time.Millisecond)
+	})
+
+	t.Run("Kill pubsub client", func(t *testing.T) {
+		// subscribe clients
+		c0 := srv.NewClient()
+		defer func() { require.NoError(t, c0.Close()) }()
+		require.NoError(t, c0.Do(ctx, "CLIENT", "SETNAME", "pubsub").Err())
+		v, err := c0.Subscribe(ctx, "foo").Receive(ctx)
+		require.NoError(t, err)
+		msg := v.(*redis.Subscription)
+		require.Equal(t, "subscribe", msg.Kind)
+		require.Equal(t, "foo", msg.Channel)
+		require.Equal(t, 1, msg.Count)
+
+		// psubscribe clients
+		c1 := srv.NewClient()
+		defer func() { require.NoError(t, c1.Close()) }()
+		require.NoError(t, c1.Do(ctx, "CLIENT", "SETNAME", "pubsub_patterns").Err())
+		v, err = c1.PSubscribe(ctx, "bar.*").Receive(ctx)
 		require.NoError(t, err)
-		require.Equal(t, r, "+PONG")
-		require.NoError(t, c.WriteArgs("PING", "PONG"))
-		r, err = c.ReadLine()
+		msg = v.(*redis.Subscription)
+		require.Equal(t, "psubscribe", msg.Kind)
+		require.Equal(t, "bar.*", msg.Channel)
+		require.Equal(t, 1, msg.Count)
+
+		// normal clients
+		c2 := srv.NewClient()
+		require.NoError(t, c2.Do(ctx, "CLIENT", "SETNAME", "normal").Err())
+		defer func() { require.NoError(t, c2.Close()) }()
+
+		require.EqualValues(t, 2, rdb.ClientKillByFilter(ctx, "type", "pubsub").Val())
+
+		// now the pubsub client should no longer be listed
+		// but normal client should not be dropped
+		t.Skip("cannot run the following assertions")
```

Comment:
The following assertions failed. Still debugging. Others work well for me.

Reply:
Fixed at https://github.com/apache/incubator-kvrocks/pull/956/commits/ba0cb9f8d6e3ae9f58ae57552c448e00af542d17.

cc @@vmihailenco you may be interested in this case:

When I'm migrating Redis's TCL tests to Go cases using go-redis, I notice that if I use go-redis's `(P)Subscribe` interface, it will hold some extra resources so that the connections underlayer cannot be released. It causes this case failed: even with `CLIENT KILL TYPE PUBSUB`, the connections remain.

Reply:
@tisonkun PubSub will try to reconnect when connection is lost. Perhaps that is what you are seeing?

Reply:
@vmihailenco Thank you! It explains the case.

Reply:
Is there a way to suppress this manner?